grails-app/services/UserProfileService.groovy

```diff
@@ -0,0 +1,29 @@
+import grails.transaction.Transactional
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.transmart.searchapp.AuthUser
+import grails.converters.JSON
+
+@Transactional
+class UserProfileService {
+
+	def springSecurityService
+
+	
+	def getAuthUserLevel() {
+		def roles = springSecurityService.getPrincipal().getAuthorities()
+
+		if (roles.contains(new SimpleGrantedAuthority("ROLE_ADMIN"))){
```

Comment:
ick - don't create new SimpleGrantedAuthority instances just to throw them away - extract the authority names and check those